source/ReproExamples/Issue514.tsx

```diff
@@ -1,66 +1,205 @@
-import { View, StyleSheet, Text } from "react-native";
-import React, { Fragment } from "react";
+import {
+  View,
+  StyleSheet,
+  Text,
+  Image,
+  ImageURISource,
+  Platform,
+} from "react-native";
+import React, { FC, Fragment } from "react";
 
-import Slider from "@react-native-community/slider";
+import Slider, { SliderProps } from "@react-native-community/slider";
 import { useState } from "react";
 
 interface TrackMarksProps {
   isTrue: boolean;
+  thumbImage?: ImageURISource;
+  StepMarker?: FC<MarkerProps> | boolean;
 }
 
-export const Issue514 = () => {
-  const [value, setValue] = useState(0);
-  const options = Array.from({ length: 7 / 1 + 1 }, (_, index) => index * 1);
+interface MarkerProps {
+  stepMarked: boolean;
+}
+
+interface CustomSliderProps extends SliderProps {
+  image?: ImageURISource;
+  stepMarker?: FC<MarkerProps> | boolean;
+  renderStepNumber?: boolean;
+}
+
+const CustomizedSlider = (props: CustomSliderProps) => {
+  const [currentValue, setCurrentValue] = useState(
+    props.value ?? props.minimumValue
+  );
+  const [width, setWidth] = useState(0);
+  const options = Array.from(
+    { length: (props.maximumValue - props.minimumValue) / props.step + 1 },
+    (_, index) => index
+  );
+
   return (
-    <View>
+    <View
+      onLayout={(event) => {
+        setWidth(event.nativeEvent.layout.width);
+      }}
+      style={[{ zIndex: 1, width: "100%" }, props.style]}
+    >
       <Slider
+        {...props}
+        onValueChange={(value) => {
+          setCurrentValue(value);
+          if (props.onValueChange) {
+            props.onValueChange(value);
+          }
+        }}
+        style={{ zIndex: 1, width: width }}
+        thumbTintColor={
+          props.thumbImage && !!props.stepMarker
+            ? "transparent"
+            : props.thumbTintColor
+        }
+        // eslint-disable-next-line
+        thumbImage={props.stepMarker ? undefined : props.thumbImage}
+      />
+      {props.stepMarker || !!props.renderStepNumber ? (
+        <View
+          pointerEvents="none"
+          style={{
+            width: width - options.length,
+            flexDirection: "row",
+            top: -25,
```

Comment:
To fix the difference between Android and iOS, the `top` should be based on `Platform`.